### ISSUE TITLE
[java] UnusedLocalVariable false positive with try-with-resource

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#5747]
**Affects PMD Version:** 7.13.0

**Rule:** [UnusedLocalVariable](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#unusedlocalvariable)

**Description:**

When using a lock, it might be used in a try-with-resources statement, so that it is automatically released. Currently this is reported as unused (as there is really no usage in the code), but removing this variable would break the program.

This was reported previously in comments in other issues:
* https://github.com/pmd/pmd/issues/5061#issuecomment-2155893287
* https://github.com/pmd/pmd/issues/4614#issue-1781294354

The workaround is documented here: https://github.com/pmd/pmd/issues/4614#issuecomment-1615979872 :
If the variable is named `ignored` or `unused`, then the rule ignores this variable.

However, naming the lock "unused", this is confusing. Maybe we can make some exceptions of unused local variables within a try-with-resources var declaration. E.g. for `java.nio.channels.FileLock`.

But maybe not. This sounds very specific. Maybe we should keep the rule as is and only make this case explicit in an example in the documentation.

**Code Sample demonstrating the issue:**

```java
try (FileOutputStream fos = new FileOutputStream(file);
                FileChannel out = fos.getChannel();
                FileLock lock = out.lock()) {            // unused variable "lock"
            final int buffsize = 4096;
            long position = 0;
            long read;
            while ((read = out.transferFrom(channel, position, buffsize)) > 0) {
                position += read;
            }
            return position;
        }

try (Closeable s = source()) {                  // unused variable "s"
  // nothing
} finally {
  sink().close();
}
```

**Expected outcome:**

PMD reports a violation at line ..., but that's wrong. That's a false positive.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
